nifi-nar-bundles/nifi-elasticsearch-bundle/nifi-elasticsearch-client-service/src/test/groovy/org/apache/nifi/elasticsearch/integration/ElasticSearch5ClientService_IT.groovy

```diff
@@ -175,6 +177,41 @@ class ElasticSearch5ClientService_IT {
         runner.assertValid()
     }
 
+    @Test
+    void testNullSuppression() {
```

Comment:
I couldn't get any of these tests to pass, looks like it expects the `messages` index to already exist, and for things like testBasicSearch it appears that it was supposed to be primed with 15 messages. I created an empty `messages` index and ran just this testNullSuppression, and got the following error (using ES 5.6.16):

```
org.apache.nifi.elasticsearch.ElasticsearchError: org.elasticsearch.client.ResponseException: POST http://localhost:9200/_bulk: HTTP/1.1 400 Bad Request
{"error":{"root_cause":[{"type":"action_request_validation_exception","reason":"Validation Failed: 1: type is missing;"}],"type":"action_request_validation_exception","reason":"Validation Failed: 1: type is missing;"},"status":400}
```

Am I doing something wrong? Couldn't find anything in the ES docs about ignoring the document type...

Reply:
The integration tests run as part of the [Maven build](https://github.com/apache/nifi/blob/main/nifi-nar-bundles/nifi-elasticsearch-bundle/nifi-elasticsearch-client-service/pom.xml#L193) in the `nifi-elasticsearch-client-service` bundle, but need the correct profiles actiating, for example:
* ES 7: `mvn -P integration-7,integration-tests clean install`
* ES 6: `mvn -P integration-6,integration-tests clean install`
* ES 5: `mvn -P integration-tests clean install`

These include the setup of indices and expected documents for the tests to run (see the [test/resources](https://github.com/apache/nifi/tree/main/nifi-nar-bundles/nifi-elasticsearch-bundle/nifi-elasticsearch-client-service/src/test/resources) script files).

Running these locally with the code from this branch using JDK 11, I get successful builds with the above commands.

I'm asuming the `_type` error you're getting above is because you're not setting the `type_name` System property if you've trying to run these tests without using the Maven build (you'd also need to setup the indices/documents in the same way as the test scripts).

Reply:
Ah yeah I was running the test from my IDE. Will try with Maven instead.

Reply:
Happy for this thread to be resolved given you mention having run the integration tests below?